### ISSUE TITLE
RFC: Apps Abstraction (hosted)

### ISSUE THREAD

**Issue by PeerRich** [calcom/cal.com#2621]
## Problem for self-hosters
- .env is getting quite big
- verification process for google, zoom and outlook is quite long and complex
- getting credentials for each app (hubspot, etc.) takes time

## Solution
- give self hoster the option to either use their own credentials - or use something like auth.cal.com as an abstraction/wrapper around all apps

## Drawbacks
- infrastructure would not be fully self hosted anymore
- need to look into each Terms of Service if/how it's possible

**Comment by leog** [comment]
I guess this means we should leverage third-party APIs to automate creating apps, verify them and configure to let self-hosters a simple way to be up and running with their Cal.com instance.

We should research if every third-party provides APIs to do that automatically, otherwise that manual work will be on our side.

**Comment by zomars** [comment]
For `.env` we could add those credentials to our database via seeder or similar strategy. @hariombalhara and I discussed this previously.

Regarding "sharing" our app installations with self-hosters would need a major refactoring. Basically we would be "Stripe Connect" but for external Calendars and Video providers.

**Comment by PeerRich** [comment]
yes. I mean https://nylas.com is providing people with an abstraction. if you are a customer of nylas, you don't need to go throught the google app submission progress

**Comment by zomars** [comment]
Seems like the logical way to go. I'm just trying to think is this could be abused in any significant way.

**Comment by PeerRich** [comment]
who would like to research this? it would solve a lot of pains for our enterprise customers to go live (skipping the verifaction part)

**Comment by zomars** [comment]
Assigning @agustif to lead and tagging @leog and myself to team up for this.

**Comment by PeerRich** [comment]
great stuff!

**Comment by PeerRich** [comment]
@agustif lets work on this after merging the API Keys PR. would be massive if we find a way to make this abstraction work for at least
- google calendar
- zoom
- outlook

since they have the longest verification process
